### PR TITLE
Refactor event store health check

### PR DESCRIPTION
**Pull request by prabhakar1983**
# stream-registry PR

Refactored the EventStoreHealthCheck : Reusing the existing validate method, and checking all the configuration rather than just "compaction"

```
kafkaTopicClient.validateTopicConfigs(eventStoreTopic.getName(), eventStoreTopic.getProperties());
```

### Added
Reusing the existing validate method, and checking all the configuration rather than just "compaction"

# PR Checklist Forms

- [x] CHANGELOG.md updated
- [x] Reviewer assigned
- [x] PR assigned (presumably to submitter)
- [x] Labels added (enhancement, bug, documentation)

### CODE REVIEW

File: core/src/main/java/com/homeaway/streamplatform/streamregistry/health/EventStoreConfigHealthCheck.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    /**
+     * validate the properties of the EventStore topic with the expected properties in the config.yaml
+     *
+     * @return Result
+     * @throws ExecutionException
+     * @throws InterruptedException
+     */
+    @Override
+    protected Result check() throws ExecutionException, InterruptedException {
+        try {
```
Comment (prabhakar1983) [inline core/src/main/java/com/homeaway/streamplatform/streamregistry/health/EventStoreConfigHealthCheck.java]: reused the `kafkaTopicClient.validateTopicConfigs` method for validating the configs

File: core/src/main/java/com/homeaway/streamplatform/streamregistry/StreamRegistryApplication.java [hunk 2]
```diff
@@ -380,8 +380,8 @@ private void initAndRegisterHealthCheck(final StreamRegistryConfiguration config
         Preconditions.checkState(managedKStreams != null, "managedKStreams cannot be null.");
         Preconditions.checkState(streamResource != null, "streamResource cannot be null.");
 
-        CompactedEventStoreHealthCheck compactedEventStoreHealthCheck = new CompactedEventStoreHealthCheck(configuration, metricRegistry);
-        environment.healthChecks().register("compactedEventStoreHealthCheck", compactedEventStoreHealthCheck);
+        EventStoreConfigHealthCheck eventStoreConfigHealthCheck = new EventStoreConfigHealthCheck(configuration, metricRegistry);
+        environment.healthChecks().register("eventStoreConfigHealthCheck", eventStoreConfigHealthCheck);
```
Comment (cricket007) [inline core/src/main/java/com/homeaway/streamplatform/streamregistry/StreamRegistryApplication.java]: Adding `"HealthCheck"` seems redundant here. Could just say `"EventStoreConfig"`

Comment (cricket007) [reply core/src/main/java/com/homeaway/streamplatform/streamregistry/StreamRegistryApplication.java]: Isn't this not backwards compatible again? 

E.g. if someone depended on checking this health check name, then we're changing the name?

File: core/src/main/java/com/homeaway/streamplatform/streamregistry/health/EventStoreConfigHealthCheck.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import lombok.extern.slf4j.Slf4j;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.health.HealthCheck;
+
+import com.homeaway.streamplatform.streamregistry.configuration.StreamRegistryConfiguration;
+
+/**
+ * StreamRegistry uses Kafka as its Data-store.
```
Comment (cricket007) [inline core/src/main/java/com/homeaway/streamplatform/streamregistry/health/EventStoreConfigHealthCheck.java]: nit: This isn't necessarily always true. It only is when the infra provider uses Kafka.

File: core/src/main/java/com/homeaway/streamplatform/streamregistry/health/EventStoreConfigHealthCheck.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+ * deleted. So, it is necessary to make sure we periodically verify the availability of topic config "cleanup.policy=compact",
+ * and monitor this metric.
+ *
+ * This HealthCheck returns "Unhealthy" if the configuration "cleanup.policy=compact" is not available for the Kafka Topic,
+ * and return the metric "app.is_event_store_kafka_topic_config_valid=2" at http://HOST:8081/private/metrics
+ *
+ */
+@Slf4j
+public class EventStoreConfigHealthCheck extends HealthCheck {
+    private static final String METRIC_IS_EVENT_STORE_KAFKA_TOPIC_CONFIG_VALID = "app.is_event_store_kafka_topic_config_valid";
```
Comment (cricket007) [inline core/src/main/java/com/homeaway/streamplatform/streamregistry/health/EventStoreConfigHealthCheck.java]: I'll bring this up again. How about `app.streamregistry.(name)` metrics?

**Comment by cricket007** [comment]
Please rebase. Seems you still have the MDC commits included here.

File: core/src/main/java/com/homeaway/streamplatform/streamregistry/health/EventStoreConfigHealthCheck.java (hunk 3 above)
Comment (prabhakar1983) [reply core/src/main/java/com/homeaway/streamplatform/streamregistry/health/EventStoreConfigHealthCheck.java]: InfraProvider at this point only supports data for Kafka Clusters.

However,  the StreamMetaData is stored in Kafka

File: core/src/main/java/com/homeaway/streamplatform/streamregistry/StreamRegistryApplication.java (hunk 2 above)
Comment (prabhakar1983) [reply core/src/main/java/com/homeaway/streamplatform/streamregistry/StreamRegistryApplication.java]: We have not CUT a release yet. so, changing the name would not cause any backward compatible issue.

File: core/src/main/java/com/homeaway/streamplatform/streamregistry/health/EventStoreConfigHealthCheck.java (hunk 4 above)
Comment (prabhakar1983) [reply core/src/main/java/com/homeaway/streamplatform/streamregistry/health/EventStoreConfigHealthCheck.java]: The reason for naming this way originally was , our DD agents converts `app.is_event_store_kafka_topic_config_valid` to `streamregistry.app.is_event_store_kafka_topic_config_valid` which pushing the metrics to DD.

Comment (prabhakar1983) [reply core/src/main/java/com/homeaway/streamplatform/streamregistry/health/EventStoreConfigHealthCheck.java]: We can change it to have `app.streamregistry.is_event_store_kafka_topic_config_valid`, and create a alert with `streamregistry.app.streamregistry.is_event_store_kafka_topic_config_valid`

**Comment by prabhakar1983** [comment]
> Please rebase. Seems you still have the MDC commits included here.

When i rebased it, the automerge removed the MDC filters, and i had to put them back.

**Comment by cricket007** [comment]
I think you have rebased against your own master fork. Not the upstream, then

File: core/src/main/java/com/homeaway/streamplatform/streamregistry/StreamRegistryApplication.java (hunk 2 above)
Comment (neoword) [reply core/src/main/java/com/homeaway/streamplatform/streamregistry/StreamRegistryApplication.java]: agreed... we haven't cut for 0.5.0 so just add it to the changed part in the CHANGELOG
(two non backward compatible changes)

**Pull request merged**